5_databases/week2/homework.sql

```diff
@@ -0,0 +1,142 @@
+USE databases_HYF;
+
+--Part 1: Working with tasks
+
+SELECT *
+FROM task;
+
+--1.1. Add a task with these attributes: title, description, created, updated, due_date, status_id, user_id
+
+INSERT INTO task (id, title, description, created, updated, due_date, status_id, user_id)
+VALUES (36, 'Search for a job', 'Update your CV, check different websites.', '2017-10-25 06:54:16', '2017-10-15 13:05:09', NULL, 1, 5);
+
+--1.2.Change the title of a task
+
+UPDATE task
+SET title = 'Ride a bike around Aarhus'
+WHERE id = 24;
+
+--1.3. Change a task due date
+
+UPDATE task
+SET due_date = '2018-12-24 20:00:00'
+WHERE id = 2;
+
+--1.4. Change a task status/Mark a task as complete
+
+SELECT *
+FROM status;
+
+UPDATE task
+SET status_id = 3
+WHERE id = 3;
+
+--1.5. Delete a task
+
+DELETE
+FROM task
+WHERE id = 11;
+
+--Part 2: School database - create a new database containing the following tables:
+
+CREATE DATABASE School_database;
+
+--Create a new database containing the following tables:
+--2.1. Class: with the columns: id, name, begins (date), ends (date)
+
+CREATE TABLE CLASS (id int NOT NULL AUTO_INCREMENT,
+                                    name VARCHAR (255),
+                                         begins DATETIME NOT NULL,
+                                                         ends DATETIME NOT NULL,
+                                                                       PRIMARY KEY (id));
+
+--2.2. Student: with the columns: id, name, email, phone, class_id (foreign key)
+
+CREATE TABLE student (id int NOT NULL,
+                             name VARCHAR (255),
+                                  email VARCHAR (255) NOT NULL,
+                                                      phone VARCHAR (255) NULL,
+                                                                          class_id int NOT NULL AUTO_INCREMENT,
+                                                                                                PRIMARY KEY (id),
+                      FOREIGN KEY (class_id) REFERENCES CLASS (id));
+
+--2.3. Create an index on the name column of the student table
+
+CREATE INDEX idx_name ON student (name);
+
+--2.4. Add a new column to the class table named status which can only have the following values: not-started, ongoing, finished (hint: enumerations)
+
+ALTER TABLE CLASS ADD status ENUM ('not-started', 'ongoing', 'finished');
+
+--Part 3: More queries
+--You should do these queries on the database hyf_lesson2
+USE databases_HYF;
+
+SELECT *
+FROM task;
+
+SELECT *
+FROM USER;
+
+SELECT *
+FROM status;
+
+--3.1. Get all the tasks assigned to users whose email ends in @spotify.com
+
+SELECT task.title,
+       user.name,
+       user.email
+FROM task
+JOIN USER ON user.id = task.user_id
```

Comment:
Task has no user_id column. I think you need to join the user_task table with user and task to execute this query.